### DISCUSSION TITLE
builder-jammy-base vs builder-jammy-buildpackless-static

### DISCUSSION THREAD

**Discussion by aleksander-dorcov-94** [paketo-buildpacks/feedback#281] [Java Team]
In the context of a java/spring-boot app.
This code is produsing a **295mb** image
```
pack build img1 \
  --builder paketobuildpacks/builder-jammy-buildpackless-static \
  --buildpack paketo-buildpacks/java \
  --path . \
  --env "BP_JVM_VERSION=21" \
  --env "BP_MAVEN_BUILD_ARGUMENTS=-Dmaven.test.skip=true --no-transfer-progress package"
```
where this one is producing a **412mb** image
```
pack build img2 \
  --builder paketobuildpacks/builder-jammy-base  \
  --path . \
  --env "BP_JVM_VERSION=21" \
  --env "BP_MAVEN_BUILD_ARGUMENTS=-Dmaven.test.skip=true --no-transfer-progress package"
```
Does this mean that the first appreach is a lot better, should we always aim for the lower size images? 
Is there any advantage or disadvantage of using the first approach?

**Comment by dmikusa** [comment]
> Does this mean that the first appreach is a lot better, should we always aim for the lower size images?
> Is there any advantage or disadvantage of using the first approach?

There are always trade-offs, so the answer is that it depends. Some things to keep in mind...

 - Smaller is better for transfer size. This may mean your app starts faster if your orchestration platform has less to download. It may save you money if your orchestrator is charging you for data transfer or includes the download of the image in the time your app is "running".
 - Smaller is better for security. The less stuff in the image, the fewer things that can potentially be vulnerable and need to be updated and the fewer things an attacker could use against you.
 - Smaller is worse for debugging. If you don't have the tools you need to debug, you may not be able to get information about the state of your app when it's having problems (for example, there's no shell in the Paketo tiny/static run images so you can't enter into those containers and investigate issues).
 - Smaller may prevent certain apps from running correctly. If your language or application needs certain shared libraries or shells out to run external binaries, those may not exist in the smaller images and thus you may not be able to run your app on those.
